src/misc/portal.ts

```diff
@@ -0,0 +1,148 @@
+import { Component } from "../component/component";
+import { VNode , patch } from "../vdom/index";
+import { xml } from "../tags";
+import { OwlEvent } from "../core/owl_event";
+import { useSubEnv } from "../hooks";
+import { portalSymbol } from "./index";
+
+/**
+ * Portal
+ *
+ * The Portal component allows to render a part of a component outside it's DOM.
+ * It is for example useful for dialogs: for css reasons, dialogs are in general
+ * placed in a specific spot of the DOM (e.g. directly in the body). With the
+ * Portal, a component can conditionally specify in its tempate that it contains
+ * a dialog, and where this dialog should be inserted in the DOM.
+ *
+ * The Portal component ensures that the communication between the content of
+ * the Portal and its parent properly works: events reaching the Portal are
+ * re-triggered on an empty <portal> node located in the parent's DOM.
+ */
+
+export class Portal extends Component<any, any> {
+  static template = xml`<portal><t t-slot="default"/></portal>`;
+  static props = {
+    target: {
+      type: String,
+    }
+  }
+
+  doTargetLookUp: Boolean = true;
```

Comment:
boolean, not Boolean. There is a difference :)

Reply:
also maybe explain what it does, like you did for other attributes